File: 1 - General/Vishal/Vishal S/Easy/Plus One/Solution.go
func plusOne(digits []int) []int {
    result := []int{}
    carry := true
    slices.Reverse(digits)
    for _,val := range digits {
        if carry {
            val += 1
            carry = (val == 10)
        }
        result = append(result, val%10)
    }
    if carry {
        result = append(result, 1)
    }
    slices.Reverse(result)
    return result
}

File: 1 - General/Vishal/Vishal S/Easy/Plus One/Solution.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        carry = True
        i = len(digits) - 1
        while i >= 0 :
            if carry:
                carry = ((digits[i] + 1) == 10)
                digits[i] = (digits[i] + 1) % 10
            else:
                break
            i -= 1
        if carry :
            digits.insert(0,1)
        return digits